charms/feast-integrator/lib/charms/feast_integrator/v0/feast_store_configuration.py

```diff
@@ -0,0 +1,382 @@
+
+        Args:
+            store_configuration (StoreConfiguration): the Feast store configuration object
+        """
+        # Validate unit is leader to send data; otherwise return
+        if not self.charm.model.unit.is_leader():
+            logger.info(
+                "StoreConfigurationProivder handled send_data event when it is not the leader."
+                "Skipping event - no data sent."
+            )
+            return
+
+        relation = self.model.get_relation(self.relation_name)
+
+        if not relation:
+            raise FeastStoreConfigurationRelationMissingError(self.relation_name)
+
+        relation_data = {k: str(v) for k, v in asdict(store_configuration).items()}
+
+        # Update relation data
+        logger.debug(f"Sending data {relation_data}")
+        relation.data[self.charm.app].update(relation_data)
+
+
+class FeastStoreConfigurationRequirer(Object):
+    """Implement the Requirer end of the Feast Configuration relation.
+
+    Attributes:
+        charm (CharmBase): the requirer application
+        relation_name (str, optional): the name of the relation
+    """
+
+    on = FeastStoreConfigurationEvents()
+
+    def __init__(self, charm: CharmBase, relation_name: Optional[str] = DEFAULT_RELATION_NAME):
+        super().__init__(charm, relation_name)
+        self.charm = charm
+        self.relation_name = relation_name
+
+        self.framework.observe(
+            self.charm.on[self.relation_name].relation_changed, self._on_relation_changed
+        )
+
+        self.framework.observe(
+            self.charm.on[self.relation_name].relation_broken, self._on_relation_broken
+        )
+
+    def _on_relation_changed(self, event: BoundEvent) -> None:
+        """Handle relation-changed event for this relation."""
+        self.on.updated.emit(event.relation)
+
+    def _on_relation_broken(self, event: BoundEvent) -> None:
+        """Handle relation-broken event for this relation."""
+        self.on.updated.emit(event.relation)
+
+    def get_feature_store_yaml(self):
+        """Generate the Feast feature_store.yaml content from a FeastConfiguration instance.
+
+        Args:
+            config (FeastConfiguration): The configuration values to populate the YAML.
+
+        Returns:
+            str: A string representation of the feature_store.yaml file.
+
+        Raises:
+            FeatureStoreConfigurationRelationDataMissingError if data is missing
+            FeatureStoreConfigurationRelationMissingError: if there is no related application
+        """
+        relation = self.model.get_relation(self.relation_name)
+
+        if not relation:
+            raise FeastStoreConfigurationRelationMissingError(self.relation_name)
+
+        relation_data = relation.data[relation.app]
+
+        if not relation_data:
+            raise FeastStoreConfigurationRelationDataMissingError(self.relation_name)
+
+        config = FeastStoreConfiguration(**relation_data)
```

Comment:
Because the __post_init__ can raise the `FeastStoreConfigurationDataInvalidError` should not we here catch it or process maybe re raise so we can then put the charm to error ?

Reply:
updated: https://github.com/canonical/feast-operators/pull/9/commits/dca54c101de85f6d4893fdbe687be5ed844cd9a9